src/main/java/seedu/address/security/AccountCredential.java

```diff
@@ -0,0 +1,52 @@
+package seedu.address.security;
+
+import java.io.Serializable;
+
+/**
+ * A serializable class that contains username and hashed password
+ */
+public class AccountCredential implements Serializable {
+    private static final String DEFAULT_USERNAME = "test";
+    private static final String DEFAULT_PASSWORD = "test";
+
+    private String userName;
+    private String hashedPassword;
+
+    public AccountCredential() {
+        this.userName = DEFAULT_USERNAME;
+        this.hashedPassword = hash(DEFAULT_PASSWORD);
+    }
+
+    public AccountCredential(String userName, String Password) {
+        this.userName = userName;
+        this.hashedPassword = hash(Password);
+    }
+
+    public String getUserName() {
+        return userName;
+    }
+
+    /**
+     * Compares
+     * @param password by hashing it first then comparing with stored hashedPassword
+     * @return boolean
+     */
+    public boolean passwordIsValid(String password) {
+        String inputHashed = hash(password);
```

Comment:
My love for 1-liners itches me to put this method as

`hashedPassword.equals(hash(password));`

Please feel free to ignore me

Reply:
1-liners can make other dev harder to understand what is it doing, and that is why I specifically labeled it as input hashed. But I see what you mean, sometimes it can be quite evident, but im being more verbose to be safe